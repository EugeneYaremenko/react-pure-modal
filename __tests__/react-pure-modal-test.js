import React from 'react';
import Modal from '../src/react-pure-modal.js';
import renderer from 'react-test-renderer';

it('Should be null without props', () => {
  const component = renderer.create(<Modal />);
  expect(component.toJSON()).toBeNull();
});

it('Should be opened and closed after click', () => {
  const component = renderer.create(<Modal isOpen />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should be closed', () => {
  const component = renderer.create(<Modal isOpen={false} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('Should show content', () => {
  const component = renderer.create(<Modal isOpen>Some content of modal</Modal>);
  expect(component.toJSON()).toMatchSnapshot();
});

it('Should show only content', () => {
  const component = renderer.create(<Modal replace isOpen>Some content of modal</Modal>);
  expect(component.toJSON()).toMatchSnapshot();
});

