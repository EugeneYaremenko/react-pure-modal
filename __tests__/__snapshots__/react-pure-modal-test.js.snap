exports[`test Should be closed 1`] = `null`;

exports[`test Should be opened and closed after click 1`] = `
<div
  className="pure-modal-backdrop "
  onClick={[Function bound handleBackdropClick]}>
  <div
    className="pure-modal  ">
    <div
      className="panel panel-default">
      <div
        className="panel-heading">
        <div
          className="close"
          onClick={[Function bound close]}>
          ×
        </div>
      </div>
      <div
        className="panel-body scrollable" />
    </div>
  </div>
</div>
`;

exports[`test Should be opened and closed after click 2`] = `null`;

exports[`test Should show content 1`] = `
<div
  className="pure-modal-backdrop "
  onClick={[Function bound handleBackdropClick]}>
  <div
    className="pure-modal  ">
    <div
      className="panel panel-default">
      <div
        className="panel-heading">
        <div
          className="close"
          onClick={[Function bound close]}>
          ×
        </div>
      </div>
      <div
        className="panel-body scrollable">
        Some content of modal
      </div>
    </div>
  </div>
</div>
`;

exports[`test Should show only content 1`] = `
<div
  className="pure-modal-backdrop "
  onClick={[Function bound handleBackdropClick]}>
  <div
    className="pure-modal  ">
    Some content of modal
  </div>
</div>
`;
